content/blog/2021-11-02-agent.md

```diff
@@ -0,0 +1,104 @@
+
+![Prometheus high level view](/assets/blog/2021-11-02/prom.png)
+
+This works great, and we have seen millions successful deployments like this over the years that process millions of series. Some of them with longer time retention like 2 years so. All, allowing to query, alert and record metrics useful for both cluster admins, as well as developers.
+
+However, the CNCF world constantly grows and evolves. With the grow of managed Kubernetes solutions and clusters created on demand within seconds we are now able to treat "clusters" as a "cattle" not "pet". In some cases solutions does not even have the cluster notion anymore e.g. [kcp](https://github.com/kcp-dev/kcp), [Google Cloud Run](https://cloud.google.com/run), [Fargate](https://aws.amazon.com/fargate/) and other serverless-like platforms.
+
+Other interesting use case that emerges is the notion of **Edge** clusters or networks. With the industries like telecommunication, automotive and IoTs adopting cloud native technologies we see more and more smaller clusters with restricted amount of resources. This is forcing all observability data to be transferred to remote, bigger counterparts as almost nothing can be stored on those remote nodes.
+
+What that means? That means monitoring data has to be somehow aggregated, presented to users and sometimes even stored on the *global* level. This is often called a **Global View** feature.
+
+Naively we could think about implementing this by putting Prometheus on global level and scrape metrics cross networks, neither push metric directly from application to central location for monitoring purposes. Let me explain why those are bad ideas:
+
+* Scrape across network is a terrible idea as we employ a huge amount of unknowns to our monitoring pipeline. The pull model allows Prometheus to know why exactly metric target has problems and when. Maybe it's down, misconfigured, restarted, too slow to give us metrics (e.g CPU saturated), not discoverable by service discovery, we don't have credentials to access or just DNS, network or whole cluster is down. By putting our scraper outside of the network we risk losing some of the information due to unreliable scrape and irregular scrape intervals. Don't do it, it's not worth it. (:
```

Comment:
"network" means anything. It's not always a terrible idea. Can we use `can be a terrible idea if it adds a huge amount of unknowns to our monitoring pipeline` ?